CONTRIBUTING.md

```diff
@@ -161,6 +161,20 @@ Cluster API maintains the most recent release/releases for all supported API and
 
 - Exceptions can be filed with maintainers and taken into consideration on a case-by-case basis.
 
+### Removal of v1alpha3 & v1alpha4 apiVersions
```

Comment:
Let's add a note above (L145-L146) linking to this paragraph for the removal plan